sample/ApiSamples/Program.cs

```diff
@@ -13,15 +13,35 @@ internal class Program
 	{
 		private static void Main(string[] args)
 		{
-			Console.WriteLine("Start");
-			SampleSpamWithCustomContextFillAll();
+			if (args.Length == 2) //in case it's started with 2 arguments we try to parse those 2 args as a TraceContext
+			{
+				Console.WriteLine($"Continue trace, traceId: {args[0]}, parentId: {args[1]}");
+				var transaction2 = Agent.Tracer.StartTransaction("Transaction2", "TestTransaction", (args[0], args[1]));
+
+				try
+				{
+					transaction2.CaptureSpan("TestSpan", "TestSpanType", () => Thread.Sleep(200));
+				}
+				finally
+				{
+					transaction2.End();
+				}
 
-			//WIP: if the process terminates the agent
-			//potentially does not have time to send the transaction to the server.
-			Thread.Sleep(1000);
+				Thread.Sleep(1000);
+				Console.WriteLine("Continue trace finished");
+			}
+			else
+			{
+				Console.WriteLine("Start");
+				PassTraceContext();
 
-			Console.WriteLine("Done");
-			Console.ReadKey();
+				//WIP: if the process terminates the agent
+				//potentially does not have time to send the transaction to the server.
+				Thread.Sleep(1000);
+
+				Console.WriteLine("Done");
+				Console.ReadKey();
+			}
 		}
 
 		public static void SampleSpamWithCustomContext()
```

Comment:
I see that `SampleSpamWithCustomContext` is not used anymore so it can be removed (or at least renamed ...Spam... :) The same goes for `SampleSpamWithCustomContextFillAll`.

Reply:
Typo fixed :) 

Most of these methods are not used, but they serve as sample code. I plan to build a menu to this app that runs all the sample scenarios and we can showcase these in Kibana. Also, these all serve as calls to the public API. This project, unlike the test projects is not marked as `internalvisible` to the `Elastic.Apm` project, and I think it's very useful to have 1 project that exercises the public agent API in a way other applications would use it. This way if we'd break a public API (e.g. make it internal) this project would catch it, since it'd be a build fail.